test/subscription_map.hpp

```diff
@@ -337,7 +337,7 @@ class single_subscription_map
     }
 
     // Remove a value at the specified subscription path
-    void remove(MQTT_NS::string_view subscription) {
+    void erase(MQTT_NS::string_view subscription) {
```

Comment:
This is similar to https://en.cppreference.com/w/cpp/container/set/erase (3).
If it is easy and cheap operation, return the number of erased elements is better.

Reply:
Yes, very cheap, just returning the result of the erase from the underlying set.

Reply:
ah, I made a mistake in the erase.

Reply:
The unit test should include the subscription is removed when it is empty in multiple_subscription_map, when all the values are removed from a subscription, the subscription can be removed completely from the map. Also the reference counting should be checked in the unit test. In my test I included checking that when all subscriptions are removed, you only have the root element left.